### PR TITLE
Add client version as header to all http requests (MERGEOK)

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

**Review by vekterli — COMMENTED** [review]
Have some questions/comments I'd like resolved first, otherwise looks good

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/Headers.java [hunk 1]
```diff
@@ -11,6 +11,8 @@
     private Headers() {
     }
 
+    public static final String CLIENT_VERSION = "Vespa-Http-Client-Version";
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/Headers.java]: The`X-` prefix is deprecated as recommendation for application specific headers. We might want to prefix it for consistency with existing headers though. See https://tools.ietf.org/html/rfc6648 / https://stackoverflow.com/a/3561399/1615280.

File: vespaclient-container-plugin/src/main/java/com/yahoo/documentapi/metrics/DocumentApiMetrics.java [hunk 2]
```diff
@@ -56,4 +59,12 @@ public void reportFailure(DocumentOperationType documentOperationType, DocumentO
         feeds.add(point);
     }
 
+    public void reportClientVersion(String version) {
+        if (version != null) {
+            feedRequests.add(versionPointCache.computeIfAbsent(version, v -> new Point(Map.of("client-version", v))));
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply vespaclient-container-plugin/src/main/java/com/yahoo/documentapi/metrics/DocumentApiMetrics.java]: Fixed

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/Headers.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vekterli) [reply vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/Headers.java]: Interesting, TIL!

File: vespaclient-container-plugin/src/main/java/com/yahoo/documentapi/metrics/DocumentApiMetrics.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bjorncs) [reply vespaclient-container-plugin/src/main/java/com/yahoo/documentapi/metrics/DocumentApiMetrics.java]: I replaced the hashmap with a proper cache from Guava.

**Review by vekterli — APPROVED** [review]
:+1:

**Comment by bjorncs** [comment]
PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**